src/app/attendance/components/AttendanceActionRender.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import ActionsMenu from "@/common/ActionMenu/ActionsMenu";
+import {
+  Trash2 as TrashIcon,
+  Edit as EditIcon,
+  XCircle as XCricleIcon,
+  CheckSquare as CheckSquareIcon,
+} from "lucide-react";
+import { MemberRole } from "@/lib/constants";
+
+interface MemberInfo {
+  shift: string;
+  role: string;
+  status: string;
+}
+interface AttendanceData {
+  shift: string;
+  status: string;
+}
+
+type Props = {
+  loginUser: MemberInfo;
+  attendanceData: AttendanceData;
+};
+
+const AttendanceActionRender = ({ loginUser, attendanceData }: Props) => {
+  const handleEditInfo = () => {
+    console.log("Editing info...");
+  };
+
+  const handleDeleteMember = () => {
+    console.log("Deleting member...");
+  };
+
+  const handleApproveRequest = () => {
+    console.log("Request Status");
+  };
+  const handleRejectRequest = () => {
+    console.log("Request Status");
+  };
+
+  const baseActions = [
+    {
+      title: "Edit Info",
+      id: 2,
+      onClick: handleEditInfo,
+      icon: <EditIcon size={16} />,
+    },
+    {
+      title: "Delete Member",
+      id: 3,
+      onClick: handleDeleteMember,
+      icon: <TrashIcon size={16} className="text-red-500" />,
+      className: "text-red-500",
+    },
+  ];
+  const ApprovelRequest = [
+    {
+      title: "Approve",
+      id: 2,
+      onClick: handleApproveRequest,
+      icon: <CheckSquareIcon size={16} />,
+    },
+    {
+      title: "Reject",
+      id: 3,
+      onClick: handleRejectRequest,
+      icon: <XCricleIcon size={16} className="text-red-500" />,
+      className: "text-red-500",
+    },
+  ];
+
+  const actionMenu = React.useMemo(() => {
+    switch (loginUser.role) {
+      case MemberRole.Member:
+        return attendanceData.status === "pending" ? [...baseActions] : [];
```

Comment:
@ghulamhafeez we have enum for `pending`

Reply:
Fixed